presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -977,6 +978,22 @@ public String quoted(RemoteTableName remoteTableName)
         return emptyMap();
     }
 
+    @Override
+    public OptionalLong delete(ConnectorSession session, ConnectorTableHandle handle)
+    {
+        QueryBuilder queryBuilder = new QueryBuilder(this);
+        JdbcTableHandle tableHandle = (JdbcTableHandle) handle;
+        try (Connection connection = connectionFactory.openConnection(session);
+                PreparedStatement preparedStatement = queryBuilder.buildDeleteSql(session, connection, tableHandle.getRemoteTableName(), tableHandle.getConstraint())) {
```

Comment:
make sure the table handle does not have any unexpected attributes, like grouping sets or synthetic columns.